MetaMask/metamask-extension
# feat(actions): refactoring + improvement of github actions

## **Description**
The purpose for these changes is to refactor the github action that checks issues template and adds labels in order to make it more generic and apply it to PRs as well.
Some code duplication has been eliminated as part of this refactoring as well.

[Same PR for Mobile.](https://github.com/MetaMask/metamask-mobile/pull/7567)

## **Manual testing steps**

1. Go on this [test repo](https://github.com/gauthierpetetin-test/repo_test/)
2. Open a PR
3. Wait for ~30s and `external-contributor` label shall appear on the PR you just opened
4. Edit the PR by deleting the whole description
5. Wait for ~30s and the `INVALID-PR-TEMPLATE` shall be added

## **Screenshots/Recordings**

https://recordit.co/5sH8eRHVmZ

### **Before**

NA

### **After**

NA

## **Related issues**

Fixes https://github.com/MetaMask/metamask-extension/issues/21275

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
